src/Symfony/Component/Form/Form.php

```diff
@@ -847,6 +847,8 @@ public function add($child, $type = null, array $options = [])
                 throw new UnexpectedTypeException($child, 'string, integer or Symfony\Component\Form\FormInterface');
             }
 
+            $child = (string) $child;
```

Comment:
The code forwarding this relies on the child name to be a string (instead of int)